File: basic/combine functions into transform/javascript/example1.js
/*
Contexto: Tienes una lista de temperaturas en grados Fahrenheit y debes convertirlas a grados Celsius.
Actualmente, tienes una función que realiza la conversión de una sola temperatura de Fahrenheit a Celsius y otra función que aplica esta conversión a todas las temperaturas en la lista.
Debes combinar estas dos funciones en una sola función de transformación.
*/
// Función para convertir una temperatura de Fahrenheit a Celsius
const convertirFahrenheitACelsius = (fahrenheit) => {
    const celsius = (fahrenheit - 32) * 5 / 9;
    return celsius;
  };

  // Función para convertir una lista de temperaturas de Fahrenheit a Celsius
  const convertirListaFahrenheitACelsius = (tempFahrenheit) => {
    const tempCelsius = tempFahrenheit.map((temperatura) => convertirFahrenheitACelsius(temperatura));
    return tempCelsius;
  };

class Temperature {
    static transformTo(type, temps) {
        const tempsConverted = temps.map((temp) => {
            if (type === "f") {
                return this.#fahrenheitToCelsius(temp)
            } else {
                return this.#celsiusToFahrenheit(temp)
            }
        });
        return tempsConverted;
    }

    static #celsiusToFahrenheit(t) {
        const fahrenheit = (t * 5 / 9) + 32;
        return fahrenheit;
    }

    static #fahrenheitToCelsius(t) {
        const celsius = (t - 32) * 5 / 9;
        return celsius;
    }
}

module.exports = {convertirListaFahrenheitACelsius, Temperature};

File: basic/combine functions into transform/python/src/example_1.py
"""
Contexto: Tienes un conjunto de calificaciones de estudiantes representado como una lista.
Debes calcular el promedio de estas calificaciones. Actualmente, tienes tres funciones separadas para calcular la suma de las calificaciones, contar el número de calificaciones y luego calcular el promedio.
Debes combinar estas tres funciones en una sola función de transformación.
"""
# Función para calcular la suma de calificaciones
def calcular_suma_calificaciones(calificaciones):
    suma = 0
    for calificacion in calificaciones:
        suma += calificacion
    return suma

# Función para contar el número de calificaciones
def contar_calificaciones(calificaciones):
    return len(calificaciones)

# Función para calcular el promedio de calificaciones
def calcular_promedio(calificaciones):
    suma = calcular_suma_calificaciones(calificaciones)
    cantidad = contar_calificaciones(calificaciones)
    promedio = suma / cantidad
    return promedio

class Profesor:
    @staticmethod
    def calcular_promedio(calificaciones: list[float]) -> float:
        sumatoria_total = sum(calificaciones)
        promedio = sumatoria_total / len(calificaciones)
        return promedio
